File: apps/backend/src/EnterpriseAgentOs.Domain/features/Agents/Interfaces/IAgentRunRepository.cs
namespace EnterpriseAgentOs.Domain.Features.Agents;

public interface IAgentRunRepository
{
    Task<AgentRunRecord> CreateAsync(AgentRunRecord run, CancellationToken ct = default);
    Task<AgentRunRecord?> GetAsync(Guid runId, CancellationToken ct = default);
    Task<IReadOnlyList<AgentRunRecord>> ListForAgentAsync(Guid agentId, Guid? parentRunId = null, CancellationToken ct = default);
    Task UpdateAsync(AgentRunRecord run, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IAgentRunRepository.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.List;
import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import com.enterpriseagentos.backendjava.domain.features.agents.models.AgentRunModel;

public interface IAgentRunRepository {
    CompletableFuture<AgentRunModel> createAsync(AgentRunModel run);
    CompletableFuture<AgentRunModel> getAsync(UUID runId);
    CompletableFuture<List<AgentRunModel>> listForAgentAsync(UUID agentId, UUID parentRunId);
    CompletableFuture<Void> updateAsync(AgentRunModel run);
}
